python/activity_stream/activity_stream.py

```diff
@@ -0,0 +1,552 @@
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sys
+import sgtk
+
+from sgtk.platform.qt import QtCore, QtGui
+
+from .ui.activity_stream_widget import Ui_ActivityStreamWidget
+
+from .widget_new_item import NewItemWidget, SimpleNewItemWidget
+from .widget_note import NoteWidget
+from .widget_value_update import ValueUpdateWidget
+from .dialog_reply import ReplyDialog
+from .data_manager import ActivityStreamDataHandler
+from .overlaywidget import SmallOverlayWidget
+ 
+class ActivityStreamWidget(QtGui.QWidget):
+    """
+    QT Widget that displays the Shotgun activity stream for an entity.
+    
+    :signal entity_requested(str, int): Fires when someone clicks an entity inside
+            the activity stream. The returned parameters are entity type and entity id.
+    :signal playback_requested(dict): Fires when someone clicks the playback url
+            on a version. Returns a shotgun dictionary with information
+            about the version.
+    """
+    
+    # max number of items to show in the activity stream.
+    MAX_STREAM_LENGTH = 20
+    entity_requested = QtCore.Signal(str, int)
+    playback_requested = QtCore.Signal(dict)   
+     
+    
+    def __init__(self, parent):
+        """
+        :param parent: QT parent object
+        :type parent: :class:`~PySide.QtGui.QWidget`
+        """
+        # first, call the base class and let it do its thing.
+        QtGui.QWidget.__init__(self, parent)
+        self._bundle = sgtk.platform.current_bundle()
+        
+        # now load in the UI that was created in the UI designer
+        self.ui = Ui_ActivityStreamWidget() 
+        self.ui.setupUi(self)
+        
+        # apply styling
+        self._load_stylesheet()
+
+        # keep an overlay for loading
+        overlay_module = self._bundle.import_module("overlay_widget")
+        self.__overlay = overlay_module.ShotgunOverlayWidget(self)
+        self.__small_overlay = SmallOverlayWidget(self)
+
+        # set insertion order into list to be bottom-up
+        self.ui.activity_stream_layout.setDirection(QtGui.QBoxLayout.BottomToTop)
+
+        # create a data manager to handle backend
+        self._data_manager = ActivityStreamDataHandler(self)
+
+        # set up signals
+        self._data_manager.note_arrived.connect(self._process_new_note)
+        self._data_manager.update_arrived.connect(self._process_new_data)
+        self._data_manager.thumbnail_arrived.connect(self._process_thumbnail)
+        self.ui.note_widget.data_updated.connect(self._on_note_submitted)
+        
+        # keep handles to all widgets to be nice to the GC
+        self._loading_widget = None
+        self._other_widgets = []
+        self._widgets = {}
```

Comment:
I think these are just activity stream widgets? Could we call this `_stream_widgets` or `_as_widgets` maybe? Then `_other_widgets` would be a little clearer too.